### PR TITLE
partial #8230

### PR DESCRIPTION
**Pull request by irvnriir**
**When merged this pull request will:**
- Add support for `curatorSelected` processing

### CODE REVIEW

File: addons/zeus/functions/fnc_ui_defendArea.sqf [hunk 1]
```diff
@@ -54,21 +57,28 @@ private _fnc_onUnload = {
...
+	if (!isNull _unit) then{ _g_sc = [_unit]; };
+
     private _radius = GETVAR(_display,GVAR(radius),100);
     private _position = getPos _logic;
 
-    [QGVAR(moduleDefendArea), [_unit, _position, _radius], _unit] call CBA_fnc_targetEvent;
+    {
+    	_unit = effectiveCommander _x;
+		[QGVAR(moduleDefendArea), [_unit, _position, _radius], _unit] call CBA_fnc_targetEvent;
+	}forEach _g_sc;
```
Comment (BaerMitUmlaut) [inline addons/zeus/functions/fnc_ui_defendArea.sqf:79]: Again, what's with the formatting?

File: addons/zeus/functions/fnc_ui_defendArea.sqf [hunk 2]
```diff
@@ -35,9 +35,12 @@ private _fnc_errorAndClose = {
 };
 
 switch (false) do {
-    case !(isNull _unit): {
+    case !(isNull _unit and (_g_sc == [])): {
         [LSTRING(NothingSelected)] call _fnc_errorAndClose;
     };
+	case (isNull _unit): {
+		_g_sc = [_unit];
+	};
```
Comment (BaerMitUmlaut) [inline addons/zeus/functions/fnc_ui_defendArea.sqf:43]: Wrong indentation.

File: addons/zeus/functions/fnc_ui_defendArea.sqf [hunk 3]
```diff
@@ -54,21 +57,28 @@ private _fnc_onUnload = {
...
+    {
+    	_unit = effectiveCommander _x;
+		[QGVAR(moduleDefendArea), [_unit, _position, _radius], _unit] call CBA_fnc_targetEvent;
+	}forEach _g_sc;
     deleteVehicle _logic;
 };
 
 _display displayAddEventHandler ["Unload", _fnc_onUnload];
-_control ctrlAddEventHandler ["ButtonClick", _fnc_onConfirm];
+_control ctrlAddEventHandler ["ButtonClick", _this + [_g_sc] call _fnc_onConfirm];
```
Comment (BaerMitUmlaut) [inline addons/zeus/functions/fnc_ui_defendArea.sqf:84]: Since in your params you expect `_g_sc` to be part of `_this`, this doesn't make any sense to me.

**Review by BaerMitUmlaut — CHANGES_REQUESTED** [review]
Congrats, not a single line that was acceptable. That's a first.

**Comment by BaerMitUmlaut** [comment]
Since none of this can be merged at this point and you didn't even make the effort of naming your PR in a sensible way, I'm closing this. Feel free to reopen it once you've got this in an acceptable state.

**Comment by irvnriir** [comment]
i'm sor that you wasted the time to describe . but this shitty and nonsense code structure already bothered me too much to put any more time in (even reading/examining it took too much effort, not even counting including the feature) . the feature is very simple, and this my example should work . include it if you don't mind .

**Comment by commy2** [comment]
This PR broke my game.

**Pull request closed without merging**